src/engine/Game.java

```diff
@@ -112,19 +115,28 @@ private void zombieTurn() {
 		
 		while (iterator.hasNext()) {
 			Zombie nextZombie = iterator.next();
-			//if a zombie has failed to move, it means it is being blocked by a Plant
-			if (!nextZombie.move(levelInfo.getRows())) {
-				nextZombie.attack(board);
-				
-				if(nextZombie.getZombieType() == ZombieTypes.EXP_ZOMBIE){ //if a exploding zombie attacks, it instantly dies
-					zombiesToRemove.add(nextZombie);
+			if(!getZomRemoveBin().contains(nextZombie))
```

Comment:
!getZomRemoveBin().contains(nextZombie) equals false when the zombie has not been called by the lawn mower. 

This is needed because written above in the Game class is List<Zombie> zombiesInGame = new LinkedList<Zombie>(board.getZombiesInGame());. We do not want move zombies that are dead

Reply:
I see very nice 👍